TASVideos/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -63,12 +66,26 @@ public static IApplicationBuilder UseMvcWithOptions(this IApplicationBuilder app
 			"form-action 'self'", // domains allowed for `<form action/>` (POST target page)
 			"frame-src 'self' https://www.youtube.com/embed/", // allow these domains in <iframe/>
 			"img-src *", // allow hotlinking images from any domain in UGC (not great)
+			/*
+			"report-to for-csp", // browsers using CSP Level 3+ will look up this key in the `Reporting-Endpoints` header and use that URI
+			$"report-uri {userAgentReportUrl}?kind=csp&csp-lvl=lteq2", // browsers from before CSP Level 3 will use this
+			*/
 			"require-trusted-types-for 'script'", // experimental, but Google seems to be pushing it: should block `HTMLScriptElement.innerHTML = "user.pwn();";`, and similarly block adding in-line scripts as attrs
 			$"script-src 'self' {string.Join(' ', trustedJsHosts)}", // `<script/>`s will be blocked unless they're from one of these domains
 			"style-src 'unsafe-inline' 'self' https://cdnjs.cloudflare.com/ajax/libs/font-awesome/", // allow `<style/>`, and `<link rel="stylesheet"/>` if it's from our domain or trusted CDN
 			"upgrade-insecure-requests", // browser should automagically replace links to any `http://tasvideos.org/...` URL (in UGC, for example) with HTTPS
 		];
 		var contentSecurityPolicyValue = string.Join("; ", cspDirectives);
+#if SHOULD_INCLUDE_STAGING_CSP

```

Comment:
Don't do these.  Is this supposed to be for different environments? If so, the env variable has what you need

Reply:
No this is to skip sending a duplicate header.

Reply:
If you want behavior that's easily togglable, we use the AppSettings object (which is conveniently already here) with configuration in the appsettings.config file. Using that also means we don't have to recompile the project for new behaviors.